ocs_ci/deployment/rosa.py

```diff
@@ -0,0 +1,177 @@
+        self.test_cluster()
+
+    def destroy(self, log_level="DEBUG"):
+        """
+        Destroy OCP cluster specific
+
+        Args:
+            log_level (str): log level openshift-installer (default: DEBUG)
+
+        """
+        ocm.destroy_cluster(self.cluster_name)
+
+
+class ROSA(CloudDeploymentBase):
+    """
+    Deployment class for ROSA.
+    """
+
+    OCPDeployment = ROSAOCP
+
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(ROSA, self).__init__()
+        ocm.download_ocm_cli()
+        rosa.download_rosa_cli()
+
+    def deploy_ocp(self, log_cli_level="DEBUG"):
+        """
+        Deployment specific to OCP cluster on a cloud platform.
+
+        Args:
+            log_cli_level (str): openshift installer's log level
+                (default: "DEBUG")
+
+        """
+        ocm.login()
+        super(ROSA, self).deploy_ocp(log_cli_level)
+
+    def check_cluster_existence(self, cluster_name_prefix):
+        """
+        Check cluster existence based on a cluster name.
+
+        Args:
+            cluster_name_prefix (str): name prefix which identifies a cluster
+
+        Returns:
+            bool: True if a cluster with the same name prefix already exists,
+                False otherwise
+
+        """
+        cluster_list = ocm.list_cluster()
+        for cluster in cluster_list:
+            name, state = cluster
+            if state != "uninstalling" and name.startswith(cluster_name_prefix):
+                return True
+        return False
+
+    def deploy_ocs(self):
+        """
+        Deployment of ODF Managed Service addon on ROSA.
+        """
+        ceph_cluster = ocp.OCP(kind="CephCluster", namespace=self.namespace)
+        try:
+            ceph_cluster.get().get("items")[0]
+            logger.warning("OCS cluster already exists")
+            return
+        except (IndexError, CommandFailed):
+            logger.info("Running OCS basic installation")
+        rosa.install_odf_addon(self.cluster_name)
+        pod = ocp.OCP(kind=constants.POD, namespace=self.namespace)
+        # Check for Ceph pods
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-mon",
+            resource_count=3,
+            timeout=600,
+        )
+        assert pod.wait_for_resource(
+            condition="Running", selector="app=rook-ceph-mgr", timeout=600
+        )
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-osd",
+            resource_count=3,
+            timeout=600,
+        )
+
+        # Verify health of ceph cluster
+        logger.info("Done validating rook resources, waiting for HEALTH_OK")
+        try:
+            ceph_health_check(namespace=self.namespace, tries=30, delay=10)
+        except CephHealthException as ex:
+            err = str(ex)
+            logger.warning(f"Ceph health check failed with {err}")
```

Comment:
Do I understand correctly that you are ignoring the ceph health here? And just writing the error?

Reply:
Right, I was not able to use original deploy_ocs function (https://github.com/red-hat-storage/ocs-ci/blob/47a5258a57a9a8b33342f76092089dc0a72ac079/ocs_ci/deployment/deployment.py#L897) so I tried to add some checks from that function. This is partial check from that function: https://github.com/red-hat-storage/ocs-ci/blob/47a5258a57a9a8b33342f76092089dc0a72ac079/ocs_ci/deployment/deployment.py#L1002 but I agree that it doesn't make sense to keep it in current state.